fix accessibility issue for text distinguishable links

ipreencekmr opened the pull request:
Accessibility issue fixed for distinguishable links from adjacent text

- Fix made for intro middleware section and doc langs section
- Markets affected are English, Indonesia and Português 
- link-underlined class was required as putting `#intro a` as selector would not be able to identify whether there is an independent link within intro section or link within paragraph. 
- Reference Issue : [https://github.com/expressjs/expressjs.com/issues/1792](https://github.com/expressjs/expressjs.com/issues/1792)

<img width="1094" alt="English" src="https://github.com/user-attachments/assets/ca1621b7-f5c5-48ca-aea2-3ed606278134" />

<img width="1124" alt="Indonesia" src="https://github.com/user-attachments/assets/a1c06773-4b4a-4b0f-88a3-dd4b85801a89" />

<img width="1140" alt="Português" src="https://github.com/user-attachments/assets/d034a569-2e7f-4b97-adc3-ba2ad70a82e6" />

ipreencekmr committed:
fix accessibility issue for text distinguishable links

ipreencekmr committed:
correct typo

ipreencekmr committed:
Merge branch 'gh-pages' into update-intro-and-doc-langs

ipreencekmr committed:
Merge branch 'gh-pages' into update-intro-and-doc-langs

bjohansebas reviewed (changes requested):
This is a great change, although I would like us to apply underlining to all links on all pages.  

Of course, there shouldn't be underlining for the header.

ipreencekmr committed:
request change to underline all links

ipreencekmr commented:
> This is a great change, although I would like us to apply underlining to all links on all pages.
> 
> Of course, there shouldn't be underlining for the header.

@bjohansebas 

request change addressed [172a6b0](https://github.com/expressjs/expressjs.com/pull/1796/commits/172a6b0510f6198296cfe6ecd9e5e0f750cea1de)

<img width="1458" alt="Screenshot 2025-02-20 at 9 08 55 AM" src="https://github.com/user-attachments/assets/2bf65f76-1b29-4024-ad4f-8e6d4b7d293b" />

<img width="420" alt="Screenshot 2025-02-20 at 9 08 01 AM" src="https://github.com/user-attachments/assets/dc71da04-111f-4385-bcdc-9a12c86840fa" />

<img width="420" alt="Screenshot 2025-02-20 at 9 08 13 AM" src="https://github.com/user-attachments/assets/a700b2b5-27ec-40c6-a1d1-1749c5e1bf97" />

<img width="419" alt="Screenshot 2025-02-20 at 9 08 36 AM" src="https://github.com/user-attachments/assets/2df90d41-3ede-40f5-96a5-b57e55ae897f" />

ipreencekmr committed:
intro boxes h3 underlining ignored

ipreencekmr commented:
underlining sponsorship ignored, 7 markets affected [b883bbd](https://github.com/expressjs/expressjs.com/pull/1796/commits/b883bbd23f741fcb5cf447368a8ee198f8e78864)

<img width="364" alt="Screenshot 2025-02-20 at 9 36 11 AM" src="https://github.com/user-attachments/assets/bd606b85-d2c2-4b46-ae2e-be00077b725c" />

chrisdel101 commented:
Noticed the gemfile.lock has been committed. Is this file necessary to this PR? I'd say remove that unless it's there on purpose.

ipreencekmr commented:
> Noticed the gemfile.lock has been committed. Is this file necessary to this PR? I'd say remove that unless it's there on purpose.

@chrisdel101 , good catch, it might have been modified while setting up the project.  And the probable reason could be Apple's M Series chip as I can see only modifications are with respect to `arm64`.

<img width="539" alt="Screenshot 2025-02-20 at 3 20 07 PM" src="https://github.com/user-attachments/assets/a3deb173-ef65-4fcf-8db5-b883e8a2b8e4" />

I have addressed your comment in [d701730](https://github.com/expressjs/expressjs.com/pull/1796/commits/d70173049e6c7ed48d1bb70b2af48e4741582a17)

ipreencekmr committed:
reset gem lock

bjohansebas commented:
I think it shouldn't have been underlined in this part on the support page
![image](https://github.com/user-attachments/assets/2dfe2de7-3c35-40a2-b37a-54c553442eb4)
https://deploy-preview-1796--expressjscom-preview.netlify.app/en/support/

ipreencekmr committed:
Merge branch 'gh-pages' into update-intro-and-doc-langs

ipreencekmr committed:
version support underlining ignored

ipreencekmr commented:
> I think it shouldn't have been underlined in this part on the support page ![image](https://private-user-images.githubusercontent.com/103585995/415281195-2dfe2de7-3c35-40a2-b37a-54c553442eb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxMDU5NjAsIm5iZiI6MTc0MDEwNTY2MCwicGF0aCI6Ii8xMDM1ODU5OTUvNDE1MjgxMTk1LTJkZmUyZGU3LTNjMzUtNDBhMi1iMzdhLTU0YzU1MzQ0MmViNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMVQwMjQxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MTEyYjNmNzE0ZTAzYzU4MzcwZWJiYWU4MjI1MjQ4ZTZlNTNjMWYyOTA1ZjQ0OWQyNTVjY2I5MzkxYmI1NzExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cwVtZOdQZGMto8zW9fHXxLgumhUm3xvCcjjBQ4JyLss) https://deploy-preview-1796--expressjscom-preview.netlify.app/en/support/

We should ensure that links are clearly identifiable to users. If we're not using an underline, we could consider alternative indicators, such as displaying a link icon on hover. Here's a reference for inspiration: [page link](https://github.com/expressjs/expressjs.com).

<img width="289" alt="Screenshot 2025-02-22 at 9 17 58 AM" src="https://github.com/user-attachments/assets/7f5b4d97-748c-4966-8101-cd98f63d922b" />

I have omitted the underlining for these support links as per the review comment addressed in [114c9f9](https://github.com/expressjs/expressjs.com/pull/1796/commits/114c9f93bb4d6f9543d0ca112368c87b7c3973c0).

ipreencekmr commented:
To maintain consistency, this page is also affected by the recent change but resolved using [4e6c214](https://github.com/expressjs/expressjs.com/pull/1796/commits/4e6c214fddaa47f7864bce3a6587451c20a80ba3) & [34f264d](https://github.com/expressjs/expressjs.com/pull/1796/commits/34f264d0cb9de4e5ff1cb3d6376977eb38f68e2d)

<img width="954" alt="Screenshot 2025-02-21 at 9 00 26 AM" src="https://github.com/user-attachments/assets/164b21ae-d640-40b6-b155-5e641508fb88" />

ipreencekmr committed:
review comment addressed

bjohansebas reviewed (approved):
LGTM, please resolve the conflicts

ipreencekmr committed:
review comment addressed and resolve conflict

ipreencekmr commented:
> LGTM, please resolve the conflicts

@bjohansebas, The conflict has been resolved, but the issue persists for `Node.js` link on the home page as the underlining is being ignored. I am working on applying a minimal-height underline to ensure it appears without affecting the overall design.

<img width="1058" alt="Screenshot 2025-02-22 at 8 02 36 AM" src="https://github.com/user-attachments/assets/51b76f8e-ec29-49c1-9e85-f7d94a7d68b0" />

ipreencekmr committed:
Node.js link underlined with thin height and offset

ipreencekmr commented:
Node.js anchor issue addressed [de9b185](https://github.com/expressjs/expressjs.com/pull/1796/commits/de9b1859ce43cc12429c8583dc8ac6570b0bcd05)

<img width="1033" alt="Screenshot 2025-02-22 at 8 59 38 AM" src="https://github.com/user-attachments/assets/176c8060-90e8-4476-aa05-a9fa565d035e" />

bjohansebas commented on zh-tw/index.md:
```diff
@@ -39,7 +39,7 @@ description: Express is a fast, unopinionated, minimalist web framework for Node
 
       <div id="sponsorship">
-          <a href="http://loopback.io/"><h2>LoopBack</h2> 使用基於 Express 的架構來開發模型驅動的應用程式。<br/>在 loopback.io 取得更多資訊。</a>
+          <h2><a href="http://loopback.io/">LoopBack</a></h2><a> 使用基於 Express 的架構來開發模型驅動的應用程式。<br/>在 loopback.io 取得更多資訊。</a>
```
```suggestion
          <h2><a href="http://loopback.io/">LoopBack</a></h2><a href="http://loopback.io/"> 使用基於 Express 的架構來開發模型驅動的應用程式。<br/>在 loopback.io 取得更多資訊。</a>
```

bjohansebas committed:
Update zh-tw/index.md

bjohansebas commented:
Thanks @ipreencekmr
